### PR TITLE
Allow SSH Over Private IP

### LINKED ISSUE #155: Tell vagrant to use internal IPs
I'm running vagrant from a GCE instance, trying to manage other CCE instances. The hangup is that vagrant is trying to SSH to the instance's public IP and Google does not support hairpinning. How do I override/config Vagrant to use the internal IP instead of the external/public one when doing things like provisioning and ssh?

### PR DESCRIPTION
**Pull request by utdrmac**
This PR completes what was started in #124 and resolves #155 . It also adds a bit of documentation to the README.

### CODE REVIEW

**Comment by utdrmac** [comment]
ping

**Comment by Temikus** [comment]
Ack, a lot of stuff happening, will test this today.
Thanks for the PR!

File: lib/vagrant-google/action/read_ssh_info.rb [hunk 1]
```diff
@@ -42,11 +42,24 @@ def read_ssh_info(google, machine)
...
+          # Get private_ip setting
+          use_private_ip = machine.provider_config.get_zone_config(zone).use_private_ip
+
+          # Default to use public ip address
+          ssh_info = {
             :host => server.public_ip_address,
             :port => 22
           }
+
+          if use_private_ip then
```
Comment (Temikus) [inline lib/vagrant-google/action/read_ssh_info.rb]: Nit: Please, fix a linter complaint.

```
λ rake lint
[DEPRECATION] `last_comment` is deprecated.  Please use `last_description` instead.
Running RuboCop...
Inspecting 49 files
...........C.....................................

Offenses:

lib/vagrant-google/action/read_ssh_info.rb:54:29: C: Do not use then for multi-line if.
          if use_private_ip then
                            ^^^^

49 files inspected, 1 offense detected
RuboCop failed!
```

**Comment by Temikus** [comment]
LGTM
Thanks @utdrmac !

@erjohnso - Please, merge.

**Comment by erjohnso** [comment]
Thanks @utdrmac and thanks @Temikus for the review.

File: lib/vagrant-google/action/read_ssh_info.rb (hunk 1 above)
Comment (utdrmac) [inline lib/vagrant-google/action/read_ssh_info.rb]: I'm not a ruby programmer; what's the correct way here?

Comment (Temikus) [inline lib/vagrant-google/action/read_ssh_info.rb]: @utdrmac No prob, I'll fix in the release patch.
Just an FYI: https://github.com/bbatsov/ruby-style-guide#no-then

Comment (utdrmac) [inline lib/vagrant-google/action/read_ssh_info.rb]: Ah. Weird. Didn't even realize 'then' was option for ifs

**Pull request merged**